tests/client/core/test_middleware.js

```diff
@@ -68,17 +70,51 @@ describe('Prefix Middleware', () => {
     assert.deepEqual(fakeRes.set.firstCall.args, ['vary', []]);
   });
 
-  it('should render a 404 when a URL exception is found', () => {
+  it('should render 404 when a localeURL exception exists', () => {
```

Comment:
Hmm... does `a localeUrlException triggers a 404` work any better? It clearly ties it to a config value. I could add a comment saying "URLs without a locale will still work as long as they aren't in this config"...